crates/projection/src/lib.rs

```diff
@@ -101,6 +102,28 @@ impl Projection for CameraMatrix {
         )
     }
 
+    fn pixel_to_robot_with_x(
+        &self,
+        pixel_coordinates: Point2<f32>,
+        x: f32,
+    ) -> Result<Point3<f32>, Error> {
+        if x <= 0.0 {
+            return Err(Error::BehindCamera);
+        }
+
+        let camera_ray = self.pixel_to_camera(pixel_coordinates);
+        let camera_ray_over_robot = self.camera_to_robot.rotation * camera_ray;
+
+        let elevation = x - self.camera_to_robot.translation.x;
+        let slope = elevation / camera_ray_over_robot.x;
```

Comment:
It took me quite a while to understand what this does.
While I think these names are correct, I would suggest variable names that better express the greater geometrical meaning.
Essentially we are intersecting a ray with an axis-aligned plane so perhaps call it something along the lines of `distance_to_plane` and `ray_scale`. Not sure about the second one, if the ray was always of length 1, I would call it ray length, but it doesn't. `ray_repetitions_to_reach_plane`?

The same applies to the `pixel_to_robot_with_z` function.